src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -210,63 +204,21 @@ abstract class UnCurry extends InfoTransform
      *    new $anon()
      *
      */
-    def transformFunction(fun: Function): Tree = {
-      fun.tpe match {
-        // can happen when analyzer plugins assign refined types to functions, e.g.
-        // (() => Int) { def apply(): Int @typeConstraint }
-        case RefinedType(List(funTp), decls) =>
-          debuglog(s"eliminate refinement from function type ${fun.tpe}")
-          fun.setType(funTp)
-        case _ =>
-          ()
-      }
-
-      deEta(fun) match {
-        // nullary or parameterless
-        case fun1 if fun1 ne fun => fun1
-        case _ =>
-          def typedFunPos(t: Tree) = localTyper.typedPos(fun.pos)(t)
-          val funParams = fun.vparams map (_.symbol)
-          def mkMethod(owner: Symbol, name: TermName, additionalFlags: FlagSet = NoFlags): DefDef =
-            gen.mkMethodFromFunction(localTyper)(fun, owner, name, additionalFlags)
-
-          def isSpecialized = {
-            forceSpecializationInfoTransformOfFunctionN
-            val specialized = specializeTypes.specializedType(fun.tpe)
-            !(specialized =:= fun.tpe)
-          }
+    def transformFunction(fun: Function): Tree =
+      // Undo eta expansion for parameterless and nullary methods
+      if (fun.vparams.isEmpty && isByNameRef(fun.body)) { noApply += fun.body ; fun.body }
+      else if (mustExpandFunction(fun)) gen.expandFunction(localTyper)(fun, inConstructorFlag)
```

Comment:
This adds `Serializable` as a parent unconditionally. Similarly, we unconditionally add this parent when using LMF.

`javac` only adds serialization support to LMF-based lambdas when the target type already inherits `Serializable`.

Not sure which way we should go here. Let's open a ticket to revisit for M5.